src/memory.c

```diff
@@ -199,14 +200,20 @@ static uint8_t memory_eeprom_crypt(const uint8_t *write_b, uint8_t *read_b,
     if (!dec) {
         goto err;
     }
-    memcpy(read_b, utils_hex_to_uint8(dec), MEM_PAGE_LEN);
+    if (read_b) {
+        memcpy(read_b, utils_hex_to_uint8(dec), MEM_PAGE_LEN);
+    }
     utils_zero(dec, dec_len);
     free(dec);
 
     utils_zero(mempass, MEM_PAGE_LEN);
     utils_clear_buffers();
     return DBB_OK;
 err:
+    if (read_b) {
+        // Randomize return value on error
+        hmac_sha256(mempass, MEM_PAGE_LEN, read_b, MEM_PAGE_LEN, read_b);
```

Comment:
So, the length of `read_b` is `MEM_PAGE_LEN` which matches the length of `hmac`, the last arg in the call to `hmac_sha256` here: it's [SHA256_DIGEST_LENGTH](https://github.com/digitalbitbox/mcu/blob/f5c25ba4030e64b5161e4f543b397c5d9db83736/src/sha2.h#L38).  But what happens if/when you copy this code to a newer version where the length of `read_b` or rather `MEM_PAGE_LEN` isn't equal to `SHA256_DIGEST_LENGTH`? I guess I'm a bit concerned this might be a "good" place for a future possible bug to sneak in.

Reply:
Good point. All the different macro lengths get a bit messy. How about something like:
```
#if (SHA256_DIGEST_LENGTH != MEM_PAGE_LEN)
#error "Incompatible macro values"
#endif
```
which will abort at compile time.

Reply:
That would be perfect! I'm always happy to fail at compile time rather than later.

Reply:
commit added